src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/TaintedDataAnalysis/TaintedDataSymbolMapExtensions.cs

```diff
@@ -43,5 +63,24 @@ public static bool IsSourceProperty(this TaintedDataSymbolMap<SourceInfo> source
 
             return false;
         }
+
+        /// <summary>
+        /// Determines if the given array is a tainted data source.
+        /// </summary>
+        /// <param name="sourceSymbolMap"></param>
+        /// <param name="arrayTypeSymbol"></param>
+        /// <returns></returns>
+        public static bool IsSourceArray(this TaintedDataSymbolMap<SourceInfo> sourceSymbolMap, IArrayTypeSymbol arrayTypeSymbol)
```

Comment:
>IsSourceArray [](start = 27, length = 13)

IsSourceConstantArray (so we don't imply all arrays are tainted data)

Reply:
But it does the constant value check outside this function, it only checks if the array symbol belongs to SourceInfo symbols. Will this new name be confusing?

Reply:
Good point.  IsSourceConstantArrayOfType()?